python/cudf/cudf/core/series.py

```diff
@@ -2599,23 +2599,9 @@ def __cuda_array_interface__(self):
 
     def repeat(self, repeats, axis=None):
         assert axis in (None, 0)
-        in_index = self.index
-        from cudf.core.multiindex import MultiIndex
-
-        if isinstance(in_index, MultiIndex):
-            in_index = RangeIndex(len(in_index))
-
-        new_index, col = libcudf.filling.repeat(
-            [in_index.as_column(), self._column], repeats
-        )
-
-        new_index = as_index(new_index)
-        if self.index.equals(new_index):
-            new_index = self.index
-        if isinstance(self.index, MultiIndex):
-            new_index = self.index.take(new_index)
-
-        return Series(data=col, index=new_index)
+        data = self._column.repeat(repeats)
+        new_index = self.index.repeat(repeats)
```

Comment:
Could technically optimize this by calling the libcudf function once with a table input constructed from the multiple columns, but we can do that in the future if it's a performance issue.

Reply:
I thought of that. But it's cleaner with the current abstraction of the index method. So i don't need to handle what kind (Index/MultiIndex) it is, right here.

Reply:
Agreed. This is a problem across a lot of methods so we should build a generalized solution to handle that in the future.